pkg/web/apps/apps_http_handler.go

```diff
@@ -45,28 +46,43 @@ func (a *httpHandler) GetApps(c echo.Context) error {
 	return c.JSON(http.StatusOK, apps)
 }
 
-// GetAppByID returns an app with the ID in JSON format from the AppService
-// TODO: Implement
-//GetAppById returns a app with the ID in JSON format from the AppService
-func (a *httpHandler) GetAppByID(c echo.Context) error {
-	apps, err := a.Service.GetApps()
+// TODO update app endpoint
+//UpdateApp returns apps by id as JSON from the AppService
+func (a *httpHandler) UpdateApp(c echo.Context) error {
 
-	if err != nil {
-		return err
+	id := c.Param("id")
+	if !isValidUUID(id) {
+		return httperrors.BadRequest(c, "Invalid id supplied")
 	}
+	// TODO create AppUpdate route
+	apps, err := a.Service.GetAppByID(id)
 
+	if err != nil {
+		return httperrors.GetHTTPResponseFromErr(c, err)
+	}
 	return c.JSON(http.StatusOK, apps)
+
 }
 
-// UpdateApp returns an app updated with the ID in JSON format from the AppService
-// TODO: Implement
-//UpdateApp returns a app updated with the ID in JSON format from the AppService
-func (a *httpHandler) UpdateApp(c echo.Context) error {
-	apps, err := a.Service.GetApps()
+// GetAppByID returns apps by id as JSON from the AppService
+func (a *httpHandler) GetAppByID(c echo.Context) error {
 
-	if err != nil {
-		return err
+	id := c.Param("id")
+	if !isValidUUID(id) {
+		return httperrors.BadRequest(c, "Invalid id supplied")
 	}
 
+	apps, err := a.Service.GetAppByID(id)
+
+	if err != nil {
+		return httperrors.GetHTTPResponseFromErr(c, err)
+	}
 	return c.JSON(http.StatusOK, apps)
+
+}
+
+// IsValidUUID helper function takes in a string and confirms is a UUID and returns bool
+func isValidUUID(id string) bool {
```

Comment:
This should be split out into a generic helper file in my opinion as it can be used in more than this file.

Reply:
Can't see this being used outside of this file , but I have no objection to a separate package for helper